### ISSUE TITLE
iOS crash：UIGraphicsBeginImageContext() failed to allocate CGBitampContext

### ISSUE THREAD

**Issue by GitHubYhb** [microsoft/clarity-apps#53]
![image](https://github.com/user-attachments/assets/079caaba-724f-456b-891e-c6f0d30ef781)

this is crash log by firebase, how to fix it ?
```
Fatal Exception: NSInternalInconsistencyException
0  CoreFoundation                 0x83f20 __exceptionPreprocess
1  libobjc.A.dylib                0x16018 objc_exception_throw
2  Foundation                     0x6de9a4 -[NSMutableDictionary(NSMutableDictionary) initWithContentsOfFile:]
3  UIKitCore                      0xc2f40 _UIGraphicsBeginImageContextWithOptions
4  Clarity                        0x13610 (符號缺漏 UUID 1cfb1ff7928a33869f4e20db41a65760)
5  Clarity                        0xf454 (符號缺漏 UUID 1cfb1ff7928a33869f4e20db41a65760)
6  Clarity                        0xb44c (符號缺漏 UUID 1cfb1ff7928a33869f4e20db41a65760)
7  Clarity                        0xa3e4 (符號缺漏 UUID 1cfb1ff7928a33869f4e20db41a65760)
8  Foundation                     0xf43e4 _startElementNs
9  libxml2.2.dylib                0x2968 xmlParseStartTag2
10 libxml2.2.dylib                0x125ec xmlParseTryOrFinish
11 libxml2.2.dylib                0x11748 xmlParseChunk
12 Foundation                     0x123e7c -[NSXMLParser parseData:]
13 Foundation                     0x124008 -[NSXMLParser parseData:]
14 Foundation                     0x123bcc -[NSXMLParser parseFromStream]
15 Clarity                        0x12e34 (符號缺漏 UUID 1cfb1ff7928a33869f4e20db41a65760)
16 Clarity                        0x9b70 (符號缺漏 UUID 1cfb1ff7928a33869f4e20db41a65760)
17 Clarity                        0x91d0 (符號缺漏 UUID 1cfb1ff7928a33869f4e20db41a65760)
18 Clarity                        0xf651c (符號缺漏 UUID 1cfb1ff7928a33869f4e20db41a65760)
19 Clarity                        0x82700 (符號缺漏 UUID 1cfb1ff7928a33869f4e20db41a65760)
20 libdispatch.dylib              0x213c _dispatch_call_block_and_release
21 libdispatch.dylib              0x3dd4 _dispatch_client_callout
22 libdispatch.dylib              0xb400 _dispatch_lane_serial_drain
23 libdispatch.dylib              0xbf30 _dispatch_lane_invoke
24 libdispatch.dylib              0x16cb4 _dispatch_root_queue_drain_deferred_wlh
25 libdispatch.dylib              0x16528 _dispatch_workloop_worker_thread
26 libsystem_pthread.dylib        0x4934 _pthread_wqthread
27 libsystem_pthread.dylib        0x10cc start_wqthread
        
```

**Comment by amralaa-MSFT** [comment]
Thanks @GitHubYhb for reporting the issue. Could you please provide more details to help expedite our troubleshooting?

**Describe the Bug**
A clear and concise description of what the bug is.

**Repro Steps**
If applicable, steps to reproduce the behavior.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Project Info**
If this is an issue happening in a specific Clarity project,

- Project ID

**Platform/Framework Info**

- Device: [e.g. iPhone 13 Pro]
- OS: [e.g. iOS 17.5]
- Application Framework [e.g. ReactNative, Native UIKit-based or Native SwiftUI-based]
- Clarity Version [e.g. 2.0.0]

**Additional Context**
Add any other context about the problem here. e.g. Nature of the issue (consistent or intermittent) or noticeable patterns

**Comment by ibradwan** [comment]
A fix should be released with our next version, so please stay tuned.

**Comment by GitHubYhb** [comment]
`UIGraphicsBeginImageContext` is deprecated at 17.0 or later, crash at iOS 17.0 or later, when size is zero.

```
UIKIT_EXTERN void     UIGraphicsBeginImageContext(CGSize size) API_DEPRECATED("Replace usage of UIGraphicsBeginImageContext with UIGraphicsImageRenderer.", ios(2.0, API_TO_BE_DEPRECATED), visionos(1.0, API_TO_BE_DEPRECATED));
```

**Comment by GitHubYhb** [comment]
> Thanks @GitHubYhb for reporting the issue. Could you please provide more details to help expedite our troubleshooting?
> 
> **Describe the Bug** A clear and concise description of what the bug is.
> 
> **Repro Steps** If applicable, steps to reproduce the behavior.
> 
> **Screenshots** If applicable, add screenshots to help explain your problem.
> 
> **Project Info** If this is an issue happening in a specific Clarity project,
> 
> * Project ID
> 
> **Platform/Framework Info**
> 
> * Device: [e.g. iPhone 13 Pro]
> * OS: [e.g. iOS 17.5]
> * Application Framework [e.g. ReactNative, Native UIKit-based or Native SwiftUI-based]
> * Clarity Version [e.g. 2.0.0]
> 
> **Additional Context** Add any other context about the problem here. e.g. Nature of the issue (consistent or intermittent) or noticeable patterns

### Bug
UIGraphicsBeginImageContext is deprecated at iOS 17.0 or later, crash when size is zero.

### Project ID
nfxf04najx

### Platform/Framework Info
OS: iOS 17 or later
Application Framework : Native
Clarity Version : cocoapods version is  `Clarity (0.2.0)`

**Comment by amralaa-MSFT** [comment]
The issue should be fixed if you upgrade to Clarity v2.1.0.
